src/enum_net.cpp

```diff
@@ -175,7 +173,7 @@ namespace libtorrent { namespace
 
 			if ((nl_hdr->nlmsg_flags & NLM_F_MULTI) == 0) break;
 
-		} while((nl_hdr->nlmsg_seq != seq) || (nl_hdr->nlmsg_pid != pid));
+		} while((int(nl_hdr->nlmsg_seq) != seq) || (int(nl_hdr->nlmsg_pid) != pid));
```

Comment:
it seems like perhaps the actual types of these fields should be the type of the parameters to the function, no? it seems like this may be redundant casting